From a22faeea871732207998624be9f250b8d8536093 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 19 Nov 2014 16:14:35 -0500
Subject: [ethernet] r8169: fix an if condition

Message-id: <1416413687-12219-5-git-send-email-ivecera@redhat.com>
Patchwork-id: 100355
O-Subject: [RHEL7.1 PATCH 04/16] r8169: fix an if condition
Bugzilla: 1165764
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

BZ: #1165764

Upstream commit(s):
commit 85911d71109d3dda8bb35515b78bcc1de6837785
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Fri Sep 19 13:40:25 2014 +0300

    r8169: fix an if condition

    There is an extra semi-colon so __rtl8169_set_features() is called every
    time.

    Fixes: 929a031dfd62 ('r8169: adjust __rtl8169_set_features')
    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Acked-by: Hayes Wang <hayeswang@realtek.com>--
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index ff19d2d..8dca7e9 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -1882,7 +1882,7 @@ static int rtl8169_set_features(struct net_device *dev,
  features &= NETIF_F_RXALL | NETIF_F_RXCSUM | NETIF_F_HW_VLAN_CTAG_RX;
 
  rtl_lock_work(tp);
- if (features ^ dev->features);
+ if (features ^ dev->features)
   __rtl8169_set_features(dev, features);
  rtl_unlock_work(tp);
 
-- 
1.7.1